airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/hooks/useFreeConnectorProgram.ts

```diff
@@ -34,16 +34,20 @@ export const useFreeConnectorProgram = () => {
   const { registerNotification } = useNotificationService();
   const { trackError } = useAppMonitoringService();
 
+  const removeStripeSuccessQuery = () => {
+    const { [STRIPE_SUCCESS_QUERY]: _, ...unrelatedSearchParams } = Object.fromEntries(searchParams);
+    setSearchParams(unrelatedSearchParams, { replace: true });
+  };
```

Comment:
Sheer curiosity here -- is there a reason we _need_ to clean this up or is this just a good practice not keeping things we don't need and keeping URLs neat?

Reply:
Largely just being cautious: if we don't clean up the query string, they'd get the whole popup confirmation experience again if they were to refresh the page, which would be weird.